package/MDAnalysis/analysis/density.py

```diff
@@ -465,7 +470,7 @@ def _prepare(self):
         grid, edges = np.histogramdd(np.zeros((1, 3)), bins=bins,
                                      range=arange, density=False)
         grid *= 0.0
-        self._grid = grid
+        self.results._grid = grid
```

Comment:
The overall grid shape needs to be determined in `__init__`, then each worker can have its own.

Reply:
Ah okay, yes that makes sense.

I think there was also an additional case, where I encountered this problem with the parallelization;
I will try to apply the same fix there to see if it works.

Other than that I added the Changelog and the docs, so should be ready to go :)